### PR TITLE
Validation ranges in open ports

### PR DESCRIPTION
**Pull request by hmunfru**
#### Reviewers
@jframos @jesuspg

#### Description
Fix issue https://jira.tid.es/browse/CLAUDIA-4847

#### Test
tested locally

### CODE REVIEW

File: rest-api/src/main/java/com/telefonica/euro_iaas/sdc/rest/validation/ProductResourceValidatorImpl.java [hunk 1]
```diff
@@ -220,6 +225,18 @@ private void validateMetadata(List<Metadata> metadatas) throws InvalidProductExc
     }
 
     private void checkPortMetadata(String port) throws InvalidProductException {
+
+        if (port.contains("-")) {
+
+            checkPort(port.substring(0, port.indexOf("-")));
+            checkPort(port.substring( port.indexOf("-")+1, port.length()));
```
Comment (flopezag) [inline rest-api/src/main/java/com/telefonica/euro_iaas/sdc/rest/validation/ProductResourceValidatorImpl.java]: check style

File: rest-api/src/main/java/com/telefonica/euro_iaas/sdc/rest/validation/ProductResourceValidatorImpl.java [hunk 2]
```diff
@@ -196,6 +196,11 @@ private void validateMetadata(List<Metadata> metadatas) throws InvalidProductExc
                 for (String port : ports) {
                     checkPortMetadata(port);
                 }
```
Comment (flopezag) [inline rest-api/src/main/java/com/telefonica/euro_iaas/sdc/rest/validation/ProductResourceValidatorImpl.java]: I think that it is not the best way to implement this method, in fact it has more than 100 lines of codes.

File: rest-api/src/main/java/com/telefonica/euro_iaas/sdc/rest/validation/ProductResourceValidatorImpl.java [hunk 3]
```diff
@@ -196,6 +196,11 @@ private void validateMetadata(List<Metadata> metadatas) throws InvalidProductExc
...
+            } else if (metadata.getKey().equals("open_ports_udp")) {
```
Comment (flopezag) [inline rest-api/src/main/java/com/telefonica/euro_iaas/sdc/rest/validation/ProductResourceValidatorImpl.java]: I assume that each metadata has key, doesn't?

File: rest-api/src/test/java/com/telefonica/euro_iaas/sdc/rest/validation/ProductResourceValidatorImplTest.java [hunk 4]
```diff
@@ -168,6 +168,42 @@ public void testValidateMetadataOPenPortError() throws Exception {
...
+
+    @Test(expected = InvalidEntityException.class)
+    public void testValidateMetadataRangeUdpError2() throws Exception {
+        String name = "t";
+        product.setName(name);
+        Metadata meta = new Metadata("open_ports_udp", "ab-100000000");
+        product.addMetadata(meta);
+        productResourceValidator.validateInsert(product);
+    }
+
```
Comment (flopezag) [inline rest-api/src/test/java/com/telefonica/euro_iaas/sdc/rest/validation/ProductResourceValidatorImplTest.java]: Why do you not add the following test:

@Test(expected = InvalidEntityException.class)
public void testValidateMetadataRangeUdpError3() throws Exception {
        String name = "t";
        product.setName(name);
        Metadata meta = new Metadata("open_ports_udp", "-23-1000");
        product.addMetadata(meta);
        productResourceValidator.validateInsert(product);
    }

@Test(expected = InvalidEntityException.class)
public void testValidateMetadataRangeUdpError4() throws Exception {
        String name = "t";
        product.setName(name);
        Metadata meta = new Metadata("open_ports_udp", "0");
        product.addMetadata(meta);
        productResourceValidator.validateInsert(product);
    }

@Test(expected = InvalidEntityException.class)
public void testValidateMetadataRangeUdpError5() throws Exception {
        String name = "t";
        product.setName(name);
        Metadata meta = new Metadata("open_ports_udp", "65535");
        product.addMetadata(meta);
        productResourceValidator.validateInsert(product);
    }

File: rest-api/src/main/java/com/telefonica/euro_iaas/sdc/rest/validation/ProductResourceValidatorImpl.java (hunk 3 above)
Comment (hmunfru) [inline rest-api/src/main/java/com/telefonica/euro_iaas/sdc/rest/validation/ProductResourceValidatorImpl.java]: yes

File: rest-api/src/test/java/com/telefonica/euro_iaas/sdc/rest/validation/ProductResourceValidatorImplTest.java (hunk 4 above)
Comment (hmunfru) [inline rest-api/src/test/java/com/telefonica/euro_iaas/sdc/rest/validation/ProductResourceValidatorImplTest.java]: done

File: rest-api/src/main/java/com/telefonica/euro_iaas/sdc/rest/validation/ProductResourceValidatorImpl.java (hunk 2 above)
Comment (hmunfru) [inline rest-api/src/main/java/com/telefonica/euro_iaas/sdc/rest/validation/ProductResourceValidatorImpl.java]: mm they are different validation...

File: rest-api/src/test/java/com/telefonica/euro_iaas/sdc/rest/validation/ProductResourceValidatorImplTest.java (hunk 4 above)
Comment (flopezag) [inline rest-api/src/test/java/com/telefonica/euro_iaas/sdc/rest/validation/ProductResourceValidatorImplTest.java]: lt 0 and gt 65535 is yet covered with your test but the limit 0 and 65535 is not clear. I suggest to put specific test in that case. in order to check that it is wrong or not.

File: rest-api/src/main/java/com/telefonica/euro_iaas/sdc/rest/validation/ProductResourceValidatorImpl.java (hunk 3 above)
Comment (flopezag) [inline rest-api/src/main/java/com/telefonica/euro_iaas/sdc/rest/validation/ProductResourceValidatorImpl.java]: NTC

File: rest-api/src/test/java/com/telefonica/euro_iaas/sdc/rest/validation/ProductResourceValidatorImplTest.java (hunk 4 above)
Comment (hmunfru) [inline rest-api/src/test/java/com/telefonica/euro_iaas/sdc/rest/validation/ProductResourceValidatorImplTest.java]: Done

Comment (flopezag) [inline rest-api/src/test/java/com/telefonica/euro_iaas/sdc/rest/validation/ProductResourceValidatorImplTest.java]: NTC

File: rest-api/src/main/java/com/telefonica/euro_iaas/sdc/rest/validation/ProductResourceValidatorImpl.java (hunk 2 above)
Comment (flopezag) [inline rest-api/src/main/java/com/telefonica/euro_iaas/sdc/rest/validation/ProductResourceValidatorImpl.java]: I mean it should be refactoring due to the method extends the maximum number of code lines. We can pass the PR but keep in mind this issue due to the check style and coding guidelines will put a comment on it.

Comment (hmunfru) [inline rest-api/src/main/java/com/telefonica/euro_iaas/sdc/rest/validation/ProductResourceValidatorImpl.java]: I have removed some ifs in the code. Check again

Comment (flopezag) [inline rest-api/src/main/java/com/telefonica/euro_iaas/sdc/rest/validation/ProductResourceValidatorImpl.java]: NTC

**Comment by flopezag** [comment]
LGTM (waiting coveralls...)

**Comment by hmunfru** [comment]
coverall https://coveralls.io/builds/2116623

**Comment by jframos** [comment]
LGTM

**Pull request merged**